### PR TITLE
EVA-2887 : handle if chromosome or scaffold is null and if similar assembly exist

### PR DESCRIPTION
**Pull request by nitin-ebi**

### CODE REVIEW

**Review by tcezard — APPROVED** [review]
looks good

**Review by apriltuesday — APPROVED** [review]

File: src/main/java/uk/ac/ebi/eva/contigalias/dus/NCBIBrowser.java [hunk 1]
```diff
@@ -87,14 +89,19 @@ public Optional<String> getGenomeReportDirectory(String accession) throws IOExce
         path += accession.substring(0, 3) + "/";
 
         String currPath = PATH_GENOMES_ALL + path;
-        FTPFile[] ftpFiles = super.listDirectories(currPath);
+        FTPFile[] ftpFiles = super.listFiles(currPath, FTPFileFilters.ALL);
 
         if (ftpFiles.length > 0) {
-            Optional<FTPFile> dir = Arrays.stream(ftpFiles).filter(it -> it.getName().contains(rawQuery)).findFirst();
+            Optional<FTPFile> dir = Arrays.stream(ftpFiles).filter(it -> it.getName().startsWith(rawQuery+"_")).findFirst();
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/contigalias/dus/NCBIBrowser.java:96]: done

File: src/main/java/uk/ac/ebi/eva/contigalias/dus/NCBIBrowser.java [hunk 2]
```diff
@@ -87,14 +89,19 @@ public Optional<String> getGenomeReportDirectory(String accession) throws IOExce
...
             if (dir.isPresent()) {
-                // path = "GCA/004/051/055/GCA_004051055.1_ASM405105v1/"
-                path += dir.get().getName() + "/";
-                return Optional.of(PATH_GENOMES_ALL + path);
+                if(dir.get().isSymbolicLink()){
+                    // symbolic link relative to current path Optional
+                    // curr_path = "/genomes/all/GCA/004/051/055/../../../../../archive/old_genbank/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh37"
+                    return Optional.of(Paths.get(currPath + dir.get().getLink()).normalize().toString() + "/");
+                }else if(dir.get().isDirectory()) {
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/contigalias/dus/NCBIBrowser.java]: done

File: src/main/java/uk/ac/ebi/eva/contigalias/dus/NCBIBrowser.java [hunk 3]
```diff
@@ -87,14 +89,19 @@ public Optional<String> getGenomeReportDirectory(String accession) throws IOExce
...
+        FTPFile[] ftpFiles = super.listFiles(currPath, FTPFileFilters.ALL);
 
         if (ftpFiles.length > 0) {
-            Optional<FTPFile> dir = Arrays.stream(ftpFiles).filter(it -> it.getName().contains(rawQuery)).findFirst();
+            Optional<FTPFile> dir = Arrays.stream(ftpFiles).filter(it -> it.getName().startsWith(rawQuery+"_")).findFirst();
             if (dir.isPresent()) {
-                // path = "GCA/004/051/055/GCA_004051055.1_ASM405105v1/"
-                path += dir.get().getName() + "/";
-                return Optional.of(PATH_GENOMES_ALL + path);
+                if(dir.get().isSymbolicLink()){
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/contigalias/dus/NCBIBrowser.java]: done

**Pull request merged**